1_fetch/src/calc_FDC.R

```diff
@@ -1,73 +1,468 @@
 #Function used to compute FDC-based metrics
+#inputs are the same as those for calc_HIT, except:
+#seasonal = TRUE will compute each of the metrics annually and seasonally for the months specified in the season_months vector
+#season_months is a numeric vector of months. Every 3 months are a season.
+#stat_type is a character equal to 'POR' for period of record metrics or 'ATS' for annual timeseries metrics.
 calc_FDCmetrics <- function(site_num, clean_daily_flow, yearType, 
-                            drainArea_tab, NE_probs, digits = 3){
+                            drainArea_tab, NE_probs, digits = 3,
+                            seasonal = FALSE, season_months = NULL,
+                            stat_type = 'POR'){
   print(site_num)
   data <- clean_daily_flow %>%
     filter(site_no == site_num)
   
-  drainArea <- drainArea_tab %>%
-    filter(site_no == site_num) %>%
-    pull(drainArea)
-  
   #Validate using the same function as EflowStats
   data <- validate_data(data, yearType)
   
+  #add column indicating groups of continuous years
+  data <- make_data_groups(data)
+  
+  #Add columns needed to compute seasonal information
+  if(seasonal){
+    if(is.null(season_months)){
+      stop('season_months must be specified for seasonal FDC-based metrics')
+    }
+    #Add column for month and season
+    data <- data %>% mutate(month_val = lubridate::month(date),
+                            season = dplyr::case_when(month_val %in% season_months[1:3] ~ 1,
+                                      month_val %in% season_months[4:6] ~ 2,
+                                      month_val %in% season_months[7:9] ~ 3,
+                                      month_val %in% season_months[10:12] ~ 4,
```

Comment:
We would have to assign a different definition of the water year based on the new seasons and check for continuous records in those water years. That is not implemented in this workflow. I think it would require a separate target that generates the data to use in this seasonal analysis.

Reply:
I decided to make a separate target for the seasonal analysis. I think it will make the workflow a little cleaner, and it's easier for us to test out using different months for these seasons